lib/BridgedRoom.js

```diff
@@ -300,4 +391,64 @@ BridgedRoom.prototype.getMatrixATime = function() {
     return this._matrixAtime;
 };
 
+BridgedRoom.prototype.lookupAndSetTeamInfo = function() {
+    if (!this._slack_bot_token) return Promise.resolve();
+
+    return rp({
+        uri: 'https://slack.com/api/team.info',
+        qs: {
+            token: this._slack_bot_token
+        },
+        json: true,
+    }).then((response) => {
+        if (!response.team) return;
+
+        if (!this._slack_team_domain !== response.team.domain) {
+            this._slack_team_domain = response.team.domain;
+            this._dirty = true;
+        }
+
+        if (!this._slack_team_id !== response.team.id) {
+            this._slack_team_id = response.team.id;
+            this._dirty = true;
+        }
+    });
+};
+
+BridgedRoom.prototype.lookupAndSetUserInfo = function() {
+    if (!this._slack_bot_token) return Promise.resolve();
+
+    return rp({
+        uri: 'https://slack.com/api/auth.test',
+        qs: {
+            token: this._slack_bot_token
+        },
+        json: true,
+    }).then((response) => {
+        console.log(response);
+        if (!response.user_id) return;
+
+        if (!this._slack_user_id !== response.user_id) {
+            this._slack_user_id = response.user_id;
+            this._dirty = true;
+        }
+
+        return rp({
+            uri: 'https://slack.com/api/users.info',
+            qs: {
+                token: this._slack_bot_token,
+                user: response.user_id,
+            },
+            json: true,
+        });
+    }).then((response) => {
+        if (!response.user && response.user.profile) return;
+
+        if (!this._slack_bot_id !== response.user.profile.bot_id) {
```

Comment:
The error came from here, because response his:
```
{ ok: false,
  error: 'missing_scope',
  needed: 'users:read',
  provided: 'identify,bot,channels:history,groups:history,team:read,files:write:user' }
```

I added "users:read" in oauth permission, and "user_change" in events, but now when I add the link, I got the message "Row is now pending-name" but the room does not show up in rooms list and link is not working.

Reply:
are you providing both the bot & user tokens? `--slack_bot_token xoxb-xxxxxxxxxx-xxxxxxxxxxxxxxxxxxxx --slack_user_token xoxp-xxxxxxxx-xxxxxxxxx-xxxxxxxx-xxxxxxxx`

Reply:
and did you have the `files:write:user` permission. I don't think you would need this, but it would help me debug if I knew how your's was set